prov/sockets/include/sock.h

```diff
@@ -923,20 +951,39 @@ enum {
 	SOCK_CONN_ACCEPT,
 	SOCK_CONN_REJECT,
 	SOCK_CONN_SHUTDOWN,
+	SOCK_CONN_FINALIZED,
 };
 
-struct sock_conn_req_handle {
+struct sock_ep_cm_op {
+	struct sock_conn_hdr hdr;
+	struct dlist_entry entry;
+	uint8_t list;
+	uint16_t done_len;
+	uint8_t is_complete;
+	struct sock_ep_cm_handle *handle;
+};
```

Comment:
Something isn't right with this.  It is mixing wire protocol structures (sock_conn_hdr) with non-wire fields (dlist_entry and handle).  Protocol structures should be kept separate from local structures unless there's a reason to copy a wire message to a local struct.